index.js

```diff
@@ -66,32 +140,53 @@ const queryDns = (version, options) => {
 };
 
 const queryHttps = (version, options) => {
-	let cancel;
-
 	const promise = (async () => {
+		let internalPromise = Promise.resolve();
+
 		try {
 			const requestOptions = {
 				family: version === 'v6' ? 6 : 4,
 				retries: 0,
 				timeout: options.timeout
 			};
 
-			const gotPromise = got(type[version].httpsUrl, requestOptions);
+			const urls = [].concat.apply(type[version].httpsUrls, options.urls || []);
 
-			cancel = gotPromise.cancel;
+			urls.forEach(url => {
+				// eslint-disable-next-line promise/prefer-await-to-then
+				internalPromise = internalPromise.then(async ip => {
+					if (ip) {
+						return ip;
+					}
 
-			let response;
-			try {
-				response = await gotPromise;
-			} catch (_) {
-				const gotBackupPromise = got(type[version].httpsFallbackUrl, requestOptions);
+					if (promise._cancelled) {
+						throw new got.CancelError();
+					}
 
-				cancel = gotBackupPromise.cancel;
+					const gotPromise = got(url, requestOptions);
+					promise.cancel = gotPromise.cancel;
 
-				response = await gotBackupPromise;
-			}
+					try {
+						const response = await gotPromise;
+
+						const ip = (response.body || '').trim();
+
+						if (!ip || !isIp[version](ip)) {
+							return null;
+						}
 
-			const ip = (response.body || '').trim();
+						return ip;
+					} catch (error) {
+						if (error instanceof got.CancelError) {
+							throw error;
+						}
+
+						return null;
```

Comment:
I prefer using `undefined` and not `null`.

Reply:
sure no problem